fix: Force posthog recording to be disabled outside cloud

tomi opened the pull request:
## Summary

The recordings are not enabled by default, but env variable enables opt-in. This changes the logic that opt-in requires also being a cloud deployment.


## Related tickets and issues

https://linear.app/n8n/issue/ADO-1659/self-hosted-data-bleeding-into-posthog

## Review / Merge checklist
- [x] PR title and summary are descriptive. **Remember, the title automatically goes into the changelog. Use `(no-changelog)` otherwise.** ([conventions](https://github.com/n8n-io/n8n/blob/master/.github/pull_request_title_conventions.md))

netroy commented on packages/cli/src/services/frontend.service.ts:
```diff
@@ -115,9 +115,9 @@ export class FrontendService {
-					'diagnostics.config.posthog.disableSessionRecording',
-				),
+				disableSessionRecording:
+					config.getEnv('diagnostics.config.posthog.disableSessionRecording') ||
```
why even keep the `disableSessionRecording` flag then? should we just remove it completely, and just use the deployment type?

tomi replied:
Excellent point. I think we could just check the domain on FE and disable it based on that

netroy commented on packages/editor-ui/src/stores/settings.store.ts:
```diff
@@ -193,6 +197,14 @@ export const useSettingsStore = defineStore(STORES.SETTINGS, {
 			return this.settings.releaseChannel === 'dev';
 		},
+		posthogSettings(): PosthogSettings {
+			const disableSessionRecording = !window.location.hostname.endsWith('n8n.cloud');
```
wouldn't this now enable on the preview instance? I think we should revert back to the deployment type check instead.

tomi replied:
Didn't think of that. Let's go with the deployment type check

tomi committed:
fix: Force posthog recording to be disabled outside cloud

The recordings are not enabled by default, but env variable enabled opt-in. This
changes the logic that opt-in requires cloud deployment.

janober commented:
Got released with [`n8n@1.25.1`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.25.1)
